### PR TITLE
Pulling in latest upstream

### PR DESCRIPTION
**Pull request by gaborcsapo**
Note: This commit is subject only to the project's license documentation and not Google's CLA

### CODE REVIEW

**Comment by yangtechen** [comment]
I got an compile error:
../../../source/daplink/drag-n-drop/vfs_user.c:532:21: error: expected ')' before 'GIT_DESCRIPTION'

But everything seems ok if I remove the "GIT_DESCRIPTION".

**Comment by yangtechen** [comment]
The function `tools/pre_build_script.py: generate_version_file` would fail when it runs `git describe HEAD`. This can be resolved if we run something like `git tag v0.9 -m "pulling upstream"`, which is weird though.

**Comment by gaborcsapo** [comment]
Raised the issue in the upstream https://github.com/ARMmbed/DAPLink/issues/857

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Build DAPLink (Linux)
 on:
   push:
     branches:
-      - experimental_compilers
+      - develop
```
Comment (eleenest) [inline .github/workflows/linux.yml:5]: I was wondering if us not having a "develop" branch is an issue, but I guess this is a no-op for us.  i.e. we don't have a develop branch, but didn't have an experimental_compilers one either.  
Good to be in sync with upstream though.

**Review by eleenest — APPROVED** [review]
Trying to review this particular commit (in the CL).  
LGTM.

**Review by eleenest — COMMENTED** [review]
Good catch.  

I believe these came from Chris Reed's fork, where he was prototyping this.  
https://github.com/flit/DAPLink
via this link: https://github.com/ARMmbed/DAPLink/issues/580#issuecomment-637079210

**Review by eleenest — COMMENTED** [review]
LGTM

**Comment by eleenest** [comment]
In the PR comment (per https://buganizer.corp.google.com/issues/195580116) can you add: 

"Note: This commit is subject only to the project's license documentation and not Google's CLA"

**Comment by eleenest** [comment]
I built and loaded the code, and verified it is up and running w the new code.  
Note: the stm32h743ii_if_crc.bin did not seem to run when I loaded it (ended up back in bootloader), but the stm32h743ii_if_crc.hex did load/run without issues.  

Note: we might want to push a tag to the github repo, or repull from upstream when a patch is in for the issue YangTe noted, but I don't think it is a blocker.

**Comment by gaborcsapo** [comment]
Yes, the CRC is required for image verification.

ARM already created a fix so I'll just cherry pick it once they're done, and then I can submit this PR.

**Pull request merged**